addon/-private/system/model/internal-model.js

```diff
@@ -855,3 +856,16 @@ if (isEnabled('ds-references')) {
   };
 
 }
+
+if (isEnabled('ds-rollback-attribute')) {
+  InternalModel.prototype.rollbackAttribute = function rollbackAttribute(attributeName) {
+    var attrs = this._attributes;
+    var dirtyKeys = Object.keys(attrs);
+    if (dirtyKeys.indexOf(attributeName) !== -1) {
+      delete this._attributes[attributeName];
+      this.send('propertyWasReset', attributeName);
+      this.record.notifyPropertyChange(attributeName);
+    }
+  }
```

Comment:
To leave all the logic to the state machine I would suggest adding a new method to the `InternalModel` which returns the latest truth for an attribute: either the inFlight value or the canonical value. Then, in `DS.Model#resetAttribute` we simply get the latest value from the internal model and `set` it. By this everything should basically behave the same as if the attribute would be reset manually to the latest value...

What do you think?